website/addons/mendeley/model.py

```diff
@@ -68,11 +72,17 @@ def _verify_client_validity(self):
         try:
             self._client.folders.list()
         except MendeleyApiException as error:
-            self._client = None
-            if error.status == 403:
-                raise HTTPError(403)
+            if error.status == 401 and 'Token has expired' in error.message:
+                try:
+                    self.refresh_oauth_key()
```

Comment:
Yes, it should be a fairly simple refactor; `return False` in the three initial failure cases, don't change the error handling, and `return True` if `self.account.<oauth_things>` are updated properly.